sycl/unittests/device/aspect-selector.cpp

```diff
@@ -0,0 +1,185 @@
+        {aspect::accelerator, PI_DEVICE_TYPE_ACC}};
+
+    if (param_value) {
+      bool DeviceTypeSet = false;
+      for (const auto &P : Types) {
+        auto It = std::find(Aspects.begin(), Aspects.end(), P.first);
+        if (It != Aspects.end()) {
+          *reinterpret_cast<pi_device_type *>(param_value) = P.second;
+          DeviceTypeSet = true;
+          break;
+        }
+      }
+
+      assert(DeviceTypeSet);
+    }
+
+    return PI_SUCCESS;
+  }
+
+  if (param_name == PI_DEVICE_INFO_PARENT_DEVICE) {
+    if (param_value)
+      *reinterpret_cast<pi_device *>(param_value) = NULL;
+
+    return PI_SUCCESS;
+  }
+
+  if (param_name == PI_DEVICE_INFO_EXTENSIONS) {
+    if (param_value_size_ret)
+      *param_value_size_ret = 0;
+    return PI_SUCCESS;
+  }
+
+  if (param_name == PI_DEVICE_INFO_PLATFORM) {
+    if (param_value)
+      *reinterpret_cast<pi_platform *>(param_value) =
+          detail::getSyclObjImpl(*Platform)->getHandleRef();
+    return PI_SUCCESS;
+  }
+
+  if (param_name == PI_DEVICE_INFO_NAME) {
+    const std::string DevName = "Test Device # " + std::to_string(DevIdx);
+    if (param_value_size_ret)
+      *param_value_size_ret = DevName.length();
+    if (param_value) {
+      size_t L = std::min(DevName.length(), param_value_size);
+      memcpy(param_value, DevName.data(), L);
+    }
+
+    return PI_SUCCESS;
+  }
+
+  return PI_INVALID_VALUE;
+}
+
+pi_result redefinedDevicesGet(pi_platform Plt, pi_device_type DeviceType,
+                              pi_uint32 NumEntries, pi_device *Devs,
+                              pi_uint32 *NumDevices) {
+  if (NumDevices)
+    *NumDevices = Available.size();
+
+  if (Devs) {
+    long MaxIdx = std::min((size_t)NumEntries, Available.size());
+
+    for (int DevIdx = 0; DevIdx < MaxIdx; ++DevIdx)
+      Devs[DevIdx] = reinterpret_cast<pi_device>(DevIdx + 1);
+  }
+
+  return PI_SUCCESS;
+}
+
+class AspectSelector : public ::testing::Test {
+public:
+  AspectSelector() = default;
+
+protected:
+  static void SetUpTestSuite() {
+    MPlt = new platform{sycl::default_selector()};
+    if (MPlt->is_host()) {
+      return;
+    }
+
+    Platform = MPlt;
+
+    MMock = new unittest::PiMock(*MPlt);
+
+    using namespace sycl::detail;
+
+    setupDefaultMockAPIs(*MMock);
+    MMock->redefine<PiApiKind::piDeviceGetInfo>(redefinedDeviceGetInfo);
+    MMock->redefine<PiApiKind::piDevicesGet>(redefinedDevicesGet);
+    MMock->redefine<PiApiKind::piDeviceRelease>(redefinedDeviceRelease);
+    MMock->redefine<PiApiKind::piDeviceRetain>(redefinedDeviceRetain);
+  }
+
+  static void TearDownTestSuite() {
+    Platform = nullptr;
+    delete MMock;
+    delete MPlt;
+  }
+
+  static platform *MPlt;
+  static unittest::PiMock *MMock;
+};
+
+platform *AspectSelector::MPlt;
+unittest::PiMock *AspectSelector::MMock;
+
+TEST_F(AspectSelector, TestPositive) {
+  if (MPlt->is_host()) {
+    GTEST_SKIP() << "This test is not supported for host\n";
+    return;
+  }
```

Comment:
Why the test is not supported for host?

Reply:
Mocking a plugin requires for plugin to be present. {{platform::get_plugin()}} is used for fetching plugin in mock infrastructure. It'll bail out if platform is host one.